src/mixins/utils.js

```diff
@@ -65,3 +66,68 @@ export const ApplyGlow = (meshObjet, baseColor, offset = 0) => {
         meshObjet.material.emissive.setHSL(hue, saturation, atomHSL.l);
     }
 };
+
+export function createRotatingGif(wave, options = {}) {
+    const ROTATION_SPEED = options.rotationSpeed || 60; // RPM
+    const frameDuration = options.frameDuration || 0.05; // seconds
+    const sampleInterval = ROTATION_SPEED * frameDuration;
+    const canvas = wave.renderer.domElement;
+    const frames = [];
+    let frameCount = 0;
+    const totalFrames = ROTATION_SPEED;
+    const { width } = canvas;
+    const { height } = canvas;
+
+    // Store original auto-rotate settings
+    const wasAutoRotating = wave.orbitControls.autoRotate;
+    const originalSpeed = wave.orbitControls.autoRotateSpeed;
+
+    // Enable rotation
+    wave.orbitControls.autoRotate = true;
+    wave.orbitControls.autoRotateSpeed = ROTATION_SPEED;
+
+    return new Promise((resolve, reject) => {
+        const captureFrame = () => {
+            wave.render();
+            frames.push(canvas.toDataURL("image/png"));
+        };
+
+        const createGif = () => {
+            console.log("Creating GIF from frames...");
+            // Restore original rotation settings
```

Comment:
The logic can be rewritten to make it simpler and using async/await

If autorotate is True - just exit and do nothing (maybe show an alert)

captureFrame can be defined on wave class itself

captureFrames should have the logic in the `if` of the animate below

createGIF should accept frames as input and be async